pytrip/spc.py

```diff
@@ -0,0 +1,547 @@
+                        db[-1].species[-1].ne = pl
+
+                    t.get_tag()
+                    if t.code == 17:
+                        pl = self.get_payload(fd, t)
+                        db[-1].species[-1].ebindata = pl
+                    if t.code == 18:  # both tags will not be present
+                        pl = self.get_payload(fd, t)
+                        db[-1].species[-1].ebindata = db[0].species[pl].ebindata
+
+                    t.get_tag()
+                    if t.code == 19:
+                        pl = self.get_payload(fd, t)
+                        db[-1].species[-1].histdata = pl
+
+                    t.get_tag()
+                    if t.code == 20:  # running cummulative sum
+                        pl = self.get_payload(fd, t)
+                        db[-1].species[-1].rcumdata = pl
+
+            self.data = db
+
+    def get_payload(self, fd, tag):
+        cnt = 1
+
+        if self.endian == 0:
+            ste = '<'  # little endian
+        else:
+            ste = '>'  # big endian
+
+        if tag.code == 1:
+            sdtype = ste + 'a' + str(tag.size)
+            print(sdtype)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0].decode("UTF-8")
+            self.filetype = payload
+            return payload
+
+        if tag.code == 2:
+            sdtype = ste + 'a' + str(tag.size)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0].decode("UTF-8")
+            self.fileversion = payload
+            return payload
+
+        if tag.code == 3:
+            sdtype = ste + 'a' + str(tag.size)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0].decode("UTF-8")
+            self.filedate = payload
+            return payload
+
+        if tag.code == 4:
+            sdtype = ste + 'a' + str(tag.size)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0].decode("UTF-8")
+            self.targname = payload
+            return payload
+
+        if tag.code == 5:
+            sdtype = ste + 'a' + str(tag.size)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0].decode("UTF-8")
+            self.projname = payload
+            return payload
+
+        if tag.code == 6:
+            sdtype = ste + 'f' + str(tag.size)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0]
+            self.energy = payload
+            return payload
+
+        if tag.code == 7:
+            sdtype = ste + 'f' + str(tag.size)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0]
+            self.peakpos = payload
+            return payload
+
+        if tag.code == 8:
+            sdtype = ste + 'f' + str(tag.size)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0]
+            self.norm = payload
+            return payload
+
+        if tag.code == 9:  # number of depth steps
```

Comment:
Just a note: I think this will probably be set to unity for SPC applied to a single voxel.

Reply:
You mean number of steps ?